crates/common/src/version.rs

```diff
@@ -0,0 +1,42 @@
+use std::{env, error::Error};
+
+pub const NIGHTLY_VERSION_WARNING_MESSAGE: &str =
+    "This is a nightly build of Foundry. It is recommended to use the latest stable version. \
+    Visit https://book.getfoundry.sh/announcements for more information. \n\
+    To mute this warning set `FOUNDRY_DISABLE_NIGHTLY_WARNING` in your environment. \n";
+
+#[allow(clippy::disallowed_macros)]
+/// Set the build version information for Foundry binaries.
+pub fn set_build_version() -> Result<(), Box<dyn Error>> {
+    // Set the short Git SHA of the latest commit.
+    let sha = env::var("VERGEN_GIT_SHA")?;
+
+    // Set the version suffix and whether the version is a nightly build.
+    // if not on a tag: <BIN> 0.3.0-dev+ba03de0019.debug
+    // if on a tag: <BIN> 0.3.0-stable+ba03de0019.release
+    let tag_name = option_env!("TAG_NAME");
```

Comment:
should use `std::env::var` instead of the env! macros when in a build script context